examples/tutorials/cnndm_summarization.py

```diff
@@ -0,0 +1,309 @@
+#
+# ::
+#
+#   from torchtext.prototype.models import T5_BASE_GENERATION
+#   transform = T5_BASE_GENERATION.transform()
+#
+
+#######################################################################
+# Dataset
+# -------
+# torchtext provides several standard NLP datasets. For a complete list, refer to the documentation at https://pytorch.org/text/stable/datasets.html.
+# These datasets are built using composable torchdata datapipes and hence support standard flow-control and mapping/transformation
+# using user defined functions and transforms. Below, we demonstrate how to pre-process the CNNDM dataset to include the prefix necessary
+# for the model to identify the task it is performing.
+#
+# The CNNDM dataset has a train, validation, and test split. Below we demo on the test split.
+#
+# .. note::
+#       Using datapipes is still currently subject to a few caveats. If you wish
+#       to extend this example to include shuffling, multi-processing, or
+#       distributed learning, please see :ref:`this note <datapipes_warnings>`
+#       for further instructions.
+
+from functools import partial
+
+from torch.utils.data import DataLoader
+from torchtext.datasets.cnndm import CNNDM
+
+batch_size = 5
+test_datapipe = CNNDM(split="test")
+task = "summarize"
+
+
+def apply_prefix(task, x):
+    return f"{task}: " + x[0], x[1]
+
+
+test_datapipe = test_datapipe.map(partial(apply_prefix, task))
+test_datapipe = test_datapipe.batch(batch_size)
+test_datapipe = test_datapipe.rows2columnar(["article", "abstract"])
+test_dataloader = DataLoader(test_datapipe, batch_size=None)
+
+#######################################################################
+# Alternately we can also use batched API (i.e apply the prefix on the whole batch)
+#
+# ::
+#
+#   def batch_prefix(task, x):
+#    return {
+#        "article": [f'{task}: ' + y for y in x["article"]],
+#        "abstract": x["abstract"]
+#    }
+#
+#   batch_size = 5
+#   test_datapipe = CNNDM(split="test")
+#   task = 'summarize'
+#
+#   test_datapipe = test_datapipe.batch(batch_size).rows2columnar(["article", "abstract"])
+#   test_datapipe = test_datapipe.map(partial(batch_prefix, task))
+#   test_dataloader = DataLoader(test_datapipe, batch_size=None)
+#
+
+######################################################################
+# Model Preparation
+# -----------------
+#
+# torchtext provides SOTA pre-trained models that can be used directly for NLP tasks or fine-tuned on downstream tasks. Below
+# we use the pre-trained T5 model with standard base configuration to perform text summarization. For additional details on
+# available pre-trained models, please refer to documentation at https://pytorch.org/text/main/models.html
+#
+#
+from torchtext.prototype.models import T5_BASE_GENERATION
+
+
+t5_base = T5_BASE_GENERATION
+transform = t5_base.transform()
+model = t5_base.get_model()
+model.to(DEVICE)
+
```

Comment:
Is it worthwhile explicitly calling `model.eval()` since we're using the model for inference. I know you mentioned that unless the `training` parameter is set to true, dropout is set to 0, but this may not be obvious to the user in this tutorial.